scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+            ax = _format_xaxis(ax, yrs)
+
+            #Set up legend
+            fig = _make_fig_legend(case_num, fig)
+
+            #Save plot
+            plot_name = plot_loc / f"{var}_{season}_TimeSeries_Mean.{plot_type}"
+            plt.savefig(plot_name, facecolor='w')
+            plt.close()
+
+            #Add plot to website (if enabled):
+            adfobj.add_website_data(plot_name, var, case,
+                                    season=season,
+                                    plot_type="TimeSeries",
+                                    multi_case=multi_case)
+        #End if (min/max)
+
+    #End for (vars)
+    #End ANN plots
+
+    #Seasonally weighted plots
+    # - DJF, MAM, JJA, SON
+    ##########################
+    print("\n  Generating time series for seasonally weighted...")
+    for var in ts_var_list_s:
+
+        #Skip variables that have levels
+        if var not in del_s:
+            print(f"\t - time series for {var}")
+            vres = res[var]
+
+            #Set plotting parameters based off whether the user wants
+            #5-yr rolling average
+            rolling = False
+
+            if 'ts' in vres:
+                if "rolling" in vres['ts']:
+                    rolling = vres['ts']['rolling']
+
+            for season in seasons:
+                fig = plt.figure(figsize=(12,8))
+                ax = fig.add_subplot(111)
+                
+                #make list to capture max/min for each case
+                y_mins = []
+                y_maxs = []
+                for case_idx, case_name in enumerate(all_case_names):
+                    #Check for baseline, and set linestyle to dashed
+                    if case_name == data_name:
+                        label=f"{base_nickname} (baseline)"
+                        marker = "--"
+                        ax.plot(yrs[case_name].astype(int),
+                                vals[var][case_name][season],
+                                marker, c='g', label=label)
+                    #Set linestyle for test cases to solid line
+                    else:
+                        label=f"{test_nicknames[case_idx]}"
+                        marker = "-"
+                        ax.plot(yrs[case_name].astype(int),
+                                vals[var][case_name][season],
+                                marker, c=colors[case_idx],label=label)
+                    #End if
+
+                    #Attempt to set custom y-ranges
+                    #Grab mins/maxes
+                    y_mins.append(np.nanmin(vals[var][case_name][season]))
+                    y_maxs.append(np.nanmax(vals[var][case_name][season]))
+                #End for (cases)
+
+                #Set Main title for subplots:
+                ax.set_title(f"Time Series {title_var}: {var} - {season}",loc="left")
+                
+                if rolling:
+                    ax.set_title(f"5-yr rolling average",loc="right")
```

Comment:
It doesn't look like a rolling average is applied to the seasonal data.  Should it be?  If not, then I would probably remove the references to the `rolling` variable both here and above.

Reply:
Yeah, after some digging, the way the seasonal data is calculated, the rolling mean isn't as straight forward as the annual case. Therefore I am setting the rolling mean feature for the annual case only for the time being.